src/main/java/TaskList.java

```diff
@@ -0,0 +1,188 @@
+                }
+                break;
+
+            case EVENT:
+                if (parts.length < 2 || parts[1].isEmpty()) {
+                    throw new BuddyException(
+                            "OOPS!!! Please include event description, start and end date or time."
+                    );
+                } else {
+                    String[] eventParts = parts[1].split("/", 3);
+                    if (eventParts.length < 3 || eventParts[1].isEmpty() || eventParts[2].isEmpty()) {
+                        throw new BuddyException(
+                                "OOPS!!! Please include event description, start and end date or time."
+                        );
+                    }
+                    String eventStart = eventParts[1].replaceFirst("from ", "").trim();
+                    String eventEnd = eventParts[2].replaceFirst("to ", "").trim();
+
+                    addTask(new Event(eventParts[0].trim(), eventStart, eventEnd, false));
+                    System.out.println("Got it. I've added this task:\n"
+                            + getTask(this.getSize() - 1));
+                    System.out.println("Now you have " + this.getSize() + " tasks in the list.");
+                }
+                break;
+            }
+        } catch (IllegalArgumentException e) {
+            throw new BuddyException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public void saveTasks() {
+        try {
+            FileWriter fileWriter = new FileWriter(filePath);
+            for (Task task : tasks) {
+                fileWriter.write(task.toSaveFileFormat() + "\n");
+            }
+            fileWriter.close();
+        } catch (IOException e) {
+            System.out.println("Error saving tasks to file: " + e.getMessage());
+        }
+    }
+
+    public void loadTasks() {
+        try {
+            File file = new File(filePath);
+            if (file.exists()) {
+                BufferedReader reader = new BufferedReader(new FileReader(file));
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    parseTask(line);
+                }
+                reader.close();
+            }
+        } catch (IOException | BuddyException e) {
+            System.out.println("Error loading tasks from file: " + e.getMessage());
+        }
+    }
+
+    private void parseTask(String line) throws BuddyException {
+        String[] parts = line.split(" \\| ");
+        String taskType = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        switch (taskType) {
+            case "T":
+                addTask(new Todo(description, isDone));
+                break;
+
+            case "D":
+                String deadlineBy = parts[3];
+                addTask(new Deadline(description, deadlineBy, isDone));
+                break;
+
+            case "E":
```

Comment:
Perhaps instead of using a singular letter to represent class types, use the entire string instead (eg: Todo, Deadline, and Event), to make the code more readable and scalable to potentially other forms of Tasks?